File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    ans = 0
    
    isVisit = [0 for _ in range(n)]
    def dfs(start_node):
        need_visit = [start_node]
        visited = [start_node]
        while need_visit:
            cur_node = need_visit.pop()
            for connected_node in range(n):
                if computers[cur_node][connected_node] and connected_node not in visited:
                    need_visit.append(connected_node)
                    visited.append(connected_node)
                    isVisit[connected_node] = 1
                    

    for node in range(n):
        if isVisit[node]:
            continue
        dfs(node)
        ans += 1
    
    return ans
            

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = new Array(n).fill(false);
    let netCnt = 0;
    
    const dfs = (node) => {
        visited[node] = true;
        for (let nextNode = 0; nextNode < n; nextNode++) {
            if (computers[node][nextNode] && !visited[nextNode]) dfs(nextNode);
        }
    };
    
    for (let node = 0; node < n; node++) {
        if (!visited[node]) {
            dfs(node);
            netCnt += 1;
        }
    }
    
    return netCnt;
}